### ISSUE TITLE
Does this code support multi-gpus?

### ISSUE THREAD

**Issue by yueruchen** [D-X-Y/AutoDL-Projects#51]
When applying the gumbel softmax trick to multi-gpus. the gumbel-softmax needs to sample noise from gaussian distribution. But if we write the noise sampling code in the `forward()` function, it may occurs that different gpu sample a different noise and activate different path. Do you considerate this case in your code?

**Comment by D-X-Y** [comment]
I think it should be fine to use different Gumbel-Softmax sampling on each GPU, which probably introduce regularization.

**Comment by yueruchen** [comment]
The problem is that simply warpping the model using DataParallel, pytorch will collect the gradient to main gpu and do back-propagation while main gpu only activates one path. Thus gradient on other paths may be wasted.

**Comment by D-X-Y** [comment]
I think PyTorch back-propagates on all GPUs, after that these gradients will be accumulated on the first one?
